subsys/bluetooth/host/hci_core.c

```diff
@@ -5372,6 +5375,26 @@ int bt_le_scan_stop(void)
 }
 #endif /* CONFIG_BT_OBSERVER */
 
+int bt_le_set_ch_map(u8_t ch_map[5])
+{
+	struct bt_hci_cp_le_set_host_chan_classif *cp;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_HOST_CHAN_CLASSIF, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+
+	memcpy(&cp->ch_map[0], &ch_map[0], 4);
+	cp->ch_map[4] = ch_map[4] & 0x1f;
+
+	dd_ch_map = (ch_map[4] >> 5) & 0x07;
```

Comment:
Perhaps BIT_MAP(3) instead of 0x07?

Reply:
Actually this feels kind of hackish to extend the API with information which doesn't pertain to the HCI command it's otherwise mapping to. I think the advertising channel selection should be part of struct bt_le_adv_opt. We can make it backwards compatible by having the value 0 map to all three advertising channels.

Reply:
This also means that there's no need to add any dd_ch_map variable anywhere.